### ISSUE TITLE
MediaRecorder not recognizing audio input channel

### ISSUE THREAD

**Issue by 12349010** [wonderunit/storyboarder#2017]
When I go to record an audio clip via MediaRecorder, Storyboarder does not recognize my audio input. I tested this with a USB microphone and a built-in microphone but was not able to come up with audio. 

The program will allow me to "record" but no audio is actually recorded. Audio recording works fine on other programs like GarageBand and Premiere Pro.

**Desktop**
 - macOS Catalina 10.15.3
 - Chrome
 - 1.18.0

**Additional context**
Possibly related to this issue, Storyboarder has not requested permission for access to my microphone. I cannot figure out how to grant this permission if it was not requested by the application.

**Comment by audionerd** [comment]
You’re trying to record audio from within Storyboarder for a board, right? Is your USB microphone set as the input device in macOS preferences?

**Comment by 12349010** [comment]
Yes. And when testing sound input levels on the MacOS device preferences sound worked. The issue seems to be localized within the Storyboarder application.

**Comment by incanus** [comment]
Same here, seeing this on Catalina. I also never get prompted for microphone access, and thus Storyboarder doesn't show up in the Privacy preference pane.

**Comment by incanus** [comment]
I'm guessing this has something to do with a lack of the proper entitlements on the Mac and luckily, the app doesn't crash but just merely fails to record audio. 

Per [here](https://github.com/wonderunit/storyboarder/blob/3982ea67c52dd565cb1a962b3f23353689384ed9/src/js/window/audio-file-control-view.js#L372), the JavaScript [`MediaRecorder`]() API is used: 

```js
    this.mediaRecorder = new MediaRecorder(
      this.userMedia._mediaStream.mediaStream,
      {
        mimeType: 'audio/webm;codec=opus'
      }
    )
  }
```

However, per [here](https://github.com/wonderunit/storyboarder/blob/3982ea67c52dd565cb1a962b3f23353689384ed9/build/entitlements.mac.plist) in the `build/entitlements.mac.plist`, the only requested entitlement is `com.apple.security.cs.allow-unsigned-executable-memory`. 

Per [this page](https://www.electronjs.org/docs/tutorial/code-signing) on Electron code signing, it looks like you might need to add `com.apple.security.device.audio-input` and possibly make a call to `systemPreferences.askForMediaAccess('microphone')`. 

I'm guessing internally, you either aren't using a Mac, or when you are, using internal builds that don't necessarily have to be code signed. Or, maybe just not recording audio. 😄

**Comment by audionerd** [comment]
@incanus good catch thanks. We could add `askForMediaAccess`, would maybe go in `prepareToRecord` so it only runs when you're about to arm an actual recording. Would you want to do a PR for that?

**Comment by incanus** [comment]
Yeah, I can give it a try. Not really working in Node and/or Electron right now, but I think I could whip it up if I can get the app standing up in the first place. Will report back.

**Comment by incanus** [comment]
For others FYI, if you download/clone the source, optionally reset to the latest release tag (`git reset --hard v2.0.1`), and then follow the directions to `npm install` & `npm start`, it starts up and doesn't have the reported issue. At least it doesn't for me, but that could be related to already-existing Terminal-related permissions on my Mac.

**Comment by MoreTacos** [comment]
I'm having the same problem.

**Comment by MoreTacos** [comment]
I compiled the app locally and got the request for permission to use audio (and the audio recording worked properly).
